modules/caddyhttp/caddyauth/caddyauth.go

```diff
@@ -64,15 +67,19 @@ func (a Authentication) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 	var user User
 	var authed bool
 	var err error
+	logger := a.logger.With(zap.Object("request", caddyhttp.LoggableHTTPRequest{Request: r}))
 	for provName, prov := range a.Providers {
 		user, authed, err = prov.Authenticate(w, r)
 		if err != nil {
-			log.Printf("[ERROR] Authenticating with %s: %v", provName, err)
+			logger.Error("could not check authentication",
+				zap.String("provider", provName),
+				zap.Error(err))
 			continue
 		}
 		if authed {
 			break
 		}
+		logger.Warn("failed authentication", zap.String("provider", provName))
```

Comment:
Ok - so I am now seeing `failed authentication` logging for the initial 401 (i.e. with no creds), and also if I enter a valid username but invalid password. But there is no logging for an invalid username.

Reply:
```
[
  {
    "level": "warn",
    "ts": 1588889542.9543042,
    "logger": "http.handlers.authentication",
    "msg": "failed authentication",
    "request": {
      "method": "GET",
      "uri": "/holly-ipc/",
      "proto": "HTTP/1.1",
      "remote_addr": "192.168.10.1:54996",
      "host": "bbjones.nz",
      "headers": {
        "Connection": [
          "Keep-Alive"
        ]
      },
      "tls": {
        "resumed": false,
        "version": 772,
        "ciphersuite": 4865,
        "proto": "",
        "proto_mutual": true,
        "server_name": "bbjones.nz"
      }
    },
    "provider": "http_basic"
  },
  {
    "level": "warn",
    "ts": 1588889550.5812442,
    "logger": "http.handlers.authentication",
    "msg": "failed authentication",
    "request": {
      "method": "GET",
      "uri": "/entry-ipc/",
      "proto": "HTTP/2.0",
      "remote_addr": "192.168.10.1:50913",
      "host": "bbjones.nz",
      "headers": {
        "Upgrade-Insecure-Requests": [
          "1"
        ],
        "User-Agent": [
          "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4136.5 Safari/537.36"
        ],
        "Sec-Fetch-User": [
          "?1"
        ],
        "Accept-Encoding": [
          "gzip, deflate, br"
        ],
        "Accept-Language": [
          "en-NZ,en-GB;q=0.9,en-US;q=0.8,en;q=0.7"
        ],
        "Dnt": [
          "1"
        ],
        "Accept": [
          "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9"
        ],
        "Sec-Fetch-Site": [
          "none"
        ],
        "Sec-Fetch-Mode": [
          "navigate"
        ],
        "Sec-Fetch-Dest": [
          "document"
        ]
      },
      "tls": {
        "resumed": false,
        "version": 772,
        "ciphersuite": 4865,
        "proto": "h2",
        "proto_mutual": true,
        "server_name": "bbjones.nz"
      }
    },
    "provider": "http_basic"
  }
]
```

Reply:
```
[
  {
    "level": "warn",
    "ts": 1588889542.9543042,
    "logger": "http.handlers.authentication",
    "msg": "failed authentication",
    "request": {
      "method": "GET",
      "uri": "/service/",
      "proto": "HTTP/1.1",
      "remote_addr": "xxx.xxx.xxx.xxx:54996",
      "host": "domain.tld",
      "headers": {
        "Connection": [
          "Keep-Alive"
        ]
      },
      "tls": {
        "resumed": false,
        "version": 772,
        "ciphersuite": 4865,
        "proto": "",
        "proto_mutual": true,
        "server_name": "domain.tld"
      }
    },
    "provider": "http_basic"
  },
  {
    "level": "warn",
    "ts": 1588889550.5812442,
    "logger": "http.handlers.authentication",
    "msg": "failed authentication",
    "request": {
      "method": "GET",
      "uri": "/service/",
      "proto": "HTTP/2.0",
      "remote_addr": "xxx.xxx.xxx.xxx:50913",
      "host": "domain.tld",
      "headers": {
        "Upgrade-Insecure-Requests": [
          "1"
        ],
        "User-Agent": [
          "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4136.5 Safari/537.36"
        ],
        "Sec-Fetch-User": [
          "?1"
        ],
        "Accept-Encoding": [
          "gzip, deflate, br"
        ],
        "Accept-Language": [
          "en-NZ,en-GB;q=0.9,en-US;q=0.8,en;q=0.7"
        ],
        "Dnt": [
          "1"
        ],
        "Accept": [
          "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9"
        ],
        "Sec-Fetch-Site": [
          "none"
        ],
        "Sec-Fetch-Mode": [
          "navigate"
        ],
        "Sec-Fetch-Dest": [
          "document"
        ]
      },
      "tls": {
        "resumed": false,
        "version": 772,
        "ciphersuite": 4865,
        "proto": "h2",
        "proto_mutual": true,
        "server_name": "domain.tld"
      }
    },
    "provider": "http_basic"
  }
]
```

Reply:
hmm - there is an error log when an invalid username is tried - but it is coming thru with "error": "crypto/bcrypt: hashedSecret too short to be a bcrypted password".

Reply:
I think this error is happening because the account lookup in `basicauth.go` (line 114) is returning nothing for an invalid username, and thus the password check on line 118 is failing with the `hashedSecret too short` message.